[Bug] UT FAIL: TestUpdateModelReplicaMetrics

### 🐛 Describe the bug

The issue is found here. https://github.com/vllm-project/aibrix/actions/runs/30157197324/job/89758710236#step:5:231

This test is not related to the PR. Seems it's introduced in earlier check ins.

```
W0726 05:16:58.642966   24666 util.go:101] environment variable PROMETHEUS_BASIC_AUTH_SECRET_NAME is not set, using default value: 
I0726 05:16:58.642973   24666 cache_metrics.go:180] Prometheus API initialized successfully
--- FAIL: TestUpdateModelReplicaMetrics (0.00s)
    cache_metrics_test.go:483: 
        	Error Trace:	/home/runner/work/aibrix/aibrix/pkg/cache/cache_metrics_test.go:483
        	Error:      	Not equal: 
        	            	expected: "prefill"
        	            	actual  : "decode"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-prefill
        	            	+decode
        	Test:       	TestUpdateModelReplicaMetrics
E0726 05:16:58.643327   24666 cache_trace.go:94] error on track request load consumption: deployment name not found on pod decode-0
E0726 05:16:58.643398   24666 cache_trace.go:94] error on track request load consumption: deployment name not found on pod decode-0
E0726 05:16:58.643472   24666 cache_trace.go:94] error on track request load consumption: deployment name not found on pod decode-0
E0726 05:16:58.643535   24666 cache_trace.go:94] error on track request load consumption: deployment name not found on pod decode-0
Running Suite: Cache Suite
==========================
Random Seed: 1785043018
Will run 53 of 53 specs
```



### Steps to Reproduce

```
make test
```

### Expected behavior

upstream main should be clean

### Environment

nightly

--- a/pkg/cache/cache_metrics_test.go
+++ b/pkg/cache/cache_metrics_test.go
@@ -480,11 +480,14 @@ func TestUpdateModelReplicaMetrics(t *testing.T) {
 
 	store.updateModelReplicaMetrics()
 	require.Len(t, emitted, 2)
-	require.Equal(t, "prefill", emitted[0]["role"])
-	require.Equal(t, "qwen3-8B", emitted[0]["model_name"])
-	require.Equal(t, "prefill-0", emitted[0]["pod"])
-	require.Equal(t, "decode", emitted[1]["role"])
-	require.Equal(t, "decode-0", emitted[1]["pod"])
+	byPod := make(map[string]map[string]string, len(emitted))
+	for _, labels := range emitted {
+		byPod[labels["pod"]] = labels
+	}
+	require.Equal(t, "prefill", byPod["prefill-0"]["role"])
+	require.Equal(t, "qwen3-8B", byPod["prefill-0"]["model_name"])
+	require.Equal(t, "decode", byPod["decode-0"]["role"])
+	require.Equal(t, "qwen3-8B", byPod["decode-0"]["model_name"])
 	require.Equal(t, 2, store.modelReplicaEmitted.Len())
 
 	store.metaPods.Delete("default/prefill-0")

